src/components/app/MenuButtons/Publish.js

```diff
@@ -378,30 +350,76 @@ const profileSharingOptions: ExplicitConnectOptions<
   },
   component: MenuButtonsPublishImpl,
 };
-const MenuButtonsPublishConnected = explicitConnect(profileSharingOptions);
+export const MenuButtonsPublish = explicitConnect(options);
+
+type DownloadSizeProps = {|
+  // The generation is a number that only increases, and does so each time a profile
+  // sanitization option changes, or a new profile is generated. It should be increased
+  // any time a new sanitized profile is generated. This is how we know to update
+  // this component.
+  +generation: number,
+  +downloadSizePromise: Promise<string>,
+|};
```

Comment:
I think you can do the same thing without `generation`, just comparing `downloadSizePromise` with its previous value in `componentDidUpdate`, and calling unwrapPromise if it's different. And no need for `getDerivedStateFromProps`.